nomad/job_endpoint_hooks.go

```diff
@@ -420,6 +418,9 @@ func (v *jobValidate) validateVaultIdentity(t *structs.Task) ([]error, error) {
 		return warnings, nil
 	}
 
+	hasTaskWID := len(vaultWIDs) > 0
+	hasDefaultWID := v.srv.config.VaultIdentityConfig(t.Vault.Cluster) != nil
```

Comment:
Moving these down allows us to take advantage of the `t.Vault == nil` conditional above and access `t.Vault.Cluster` directly.